src/lido/LidoOracle.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+// types
+import {ISTETH} from "./ISTETH.sol";
+import {IWSTETH} from "./IWSTETH.sol";
+import {IOracle} from "../interfaces/IOracle.sol";
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+
+contract LidoOracle is IOracle {
+    ISTETH public immutable STETH;
+    IWSTETH public immutable WSTETH;
+    uint256 public immutable STETH_SCALAR;
+    uint256 public immutable WSTETH_SCALAR;
+
+    constructor(address wsteth_) {
+        // configure wsteth constants
+        WSTETH = IWSTETH(wsteth_);
+        WSTETH_SCALAR = 10 ** IERC20(address(wsteth_)).decimals();
+
+        // configure steth constants
+        STETH = ISTETH(WSTETH.stETH());
+        STETH_SCALAR = 10 ** STETH.decimals();
+    }
+
+    /// @notice Returns the value of baseAmount of baseAsset in quoteAsset terms.
+    /// @param base The asset that the user needs to know the value for.
+    /// @param quote The asset in which the user needs to value the `base`.
+    /// @param baseAmount The amount of `base` that the user wants to know the value in `quote` for.
+    /// @return quoteAmount The amount of `quote` that has the same value as `baseAmount`.
+    function valueOf(address base, address quote, uint256 baseAmount) external view returns (uint256 quoteAmount) {
+        if (base == address(STETH) && quote == address(WSTETH)) {
+            // value of given stETH amount, in terms of wstETH
+            return STETH.getSharesByPooledEth(baseAmount);
+        } else if (base == address(WSTETH) && quote == address(STETH)) {
+            // value of given wstETH amount, in terms of stETH
+            return STETH.getPooledEthByShares(baseAmount);
```

Comment:
And then this should be `WSTETH.getStEthByWstEth(baseAmount)`.

Reply:
it's a cheaper alternative to do the same, `WSTETH.getStEthByWstEth` just calls `STETH. getPooledEthByShares` internally

refer: [@totomanov's comment](https://github.com/alcueca/oracles/pull/2#issuecomment-2016760007)

same for the corresponding priceOf call